### PR TITLE
mysql tuning

### PR DESCRIPTION
**Pull request by justinjpacheco**

### CODE REVIEW

File: chef/cookbooks/bcpc/attributes/mysql.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+default['bcpc']['mysql']['repo']['enabled'] = true
+default['bcpc']['mysql']['repo']['url'] = 'http://repo.percona.com/apt'
+default['bcpc']['mysql']['repo']['key'] = 'mysql/release.key'
+
+# fqdn of mysql server
+default['bcpc']['mysql']['host'] = 'primary.mysql.service.consul'
+default['bcpc']['mysql']['service_hostname'] = 'primary.mysql.service.consul'
+
+#innodb_buffer_pool_instances: 16
```
Comment (tj90241) [inline chef/cookbooks/bcpc/attributes/mysql.rb:13]: 🤔 ?

File: chef/cookbooks/bcpc/attributes/mysql.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+default['bcpc']['mysql']['innodb_read_io_threads'] = 8
+default['bcpc']['mysql']['innodb_write_io_threads'] = 8
+default['bcpc']['mysql']['innodb_thread_concurrency'] = 80
+default['bcpc']['mysql']['join_buffer_size'] = '128M'
+default['bcpc']['mysql']['max_connections'] = 8192
+default['bcpc']['mysql']['max_heap_table_size'] = '128M'
+default['bcpc']['mysql']['sort_buffer_size'] = '1M'
+default['bcpc']['mysql']['table_open_cache'] = 5120
+default['bcpc']['mysql']['table_open_cache_instances'] = 8
+default['bcpc']['mysql']['thread_cache_size'] = 1024
```
Comment (tj90241) [inline chef/cookbooks/bcpc/attributes/mysql.rb:31]: `nil` check for this still desired in the corresponding conf file now that it's unconditionally defined?

File: chef/cookbooks/bcpc/attributes/mysql.rb (hunk 1 above)
Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/mysql.rb:13]: perfectly reasonable and helpful comment

Comment (comay) [reply chef/cookbooks/bcpc/attributes/mysql.rb:13]: I had the same reaction as Tyler. 😉 It seems the original comment made more sense?

File: chef/cookbooks/bcpc/attributes/mysql.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+default['bcpc']['mysql']['innodb_buffer_pool_instances'] = 1
+default['bcpc']['mysql']['innodb_buffer_pool_size'] = '128M'
+default['bcpc']['mysql']['innodb_flush_method'] = 'O_DIRECT_NO_FSYNC'
+default['bcpc']['mysql']['innodb_io_capacity'] = 2048
+default['bcpc']['mysql']['innodb_log_buffer_size'] = '256M'
+default['bcpc']['mysql']['innodb_log_file_size'] = '128M'
+default['bcpc']['mysql']['innodb_log_files_in_group'] = 4
+default['bcpc']['mysql']['innodb_read_io_threads'] = 8
+default['bcpc']['mysql']['innodb_write_io_threads'] = 8
+default['bcpc']['mysql']['innodb_thread_concurrency'] = 80
```
Comment (comay) [inline chef/cookbooks/bcpc/attributes/mysql.rb:24]: Does it make sense to "block" the `innodb` variables under `['bcpc']['mysql']['innodb']`?

Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/mysql.rb:24]: not sure how that block would work. everything under mysql is related to its configuration, are you suggesting we would have `wsrep` blocks as well?

Comment (comay) [reply chef/cookbooks/bcpc/attributes/mysql.rb:24]: Not necessarily - it just seemed there are a bunch of `innodb` prefixed parameters so perhaps putting them under an attribute subblock might make things more readable but it's purely semantics or syntactic sugar.

**Comment by comay** [comment]
Generally with these tuning parameters, is there any particular concern running with these in a virtual environment or are they safe to use even in memory/CPU constrained environments such as are minimally configured `1h1w`?

**Comment by justinjpacheco** [comment]
cpu usage seems to be lower: it jumps between 4 - 10 % to a spike of 30%. prior to the update it was stuck at 30%

memory usage

(before)
root@r1n1:/etc/mysql# free
              total        used        free      shared  buff/cache   available
Mem:       32940464     5485276    17327668       20152    10127520    27327128
Swap:      17780728           0    17780728

(after)
root@r1n1:/etc/mysql# free
              total        used        free      shared  buff/cache   available
Mem:       32940464     5180616    17228396       20216    10531452    27631724
Swap:      17780728           0    17780728

File: chef/cookbooks/bcpc/attributes/mysql.rb (hunk 2 above)
Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/mysql.rb:29]: good point

**Review by comay — DISMISSED** [review]

**Review by tj90241 — DISMISSED** [review]

**Review by tj90241 — APPROVED** [review]

**Review by comay — APPROVED** [review]

**Pull request merged**